files/usr/share/jeos-firstboot/jeos-firstboot-functions

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+stty_size() {
+	set -- `stty size`; LINES=$1; COLUMNS=$2
+	# stty size can return zero when not ready or
+	# its a serial console
+	if [ "$COLUMNS" = "0" -o "$LINES" = "0" ]; then
+		LINES=24
+		COLUMNS=80
+	fi
+}
+stty_size
+
+result=
+list=
+password=''
+
+let dh_menu=LINES-15
+let dh_text=LINES-5
+
+call_module_config() {
+        local module="$1"
+        module_function="${module}_jeos_config"
+        [ "$(type -t "${module_function}")" = "function" ] || return 1
+        "${module_function}" && true # To not trigger errexit
+        ret=$?
+        [ $ret -eq 0 ] || return $ret
+}
+
+kmscon_available() {
+	# kmscon itself is installed
+	kmscon --help >/dev/null 2>&1 || return 1
+	# At least one monospace font is available
+	[ -n "$(fc-match "monospace" 2>/dev/null)" ] || return 1
+
+	return 0
+}
+
+fbiterm_available() {
+	# fbiterm itself is installed
+	fbiterm --help >/dev/null 2>&1 || return 1
+	# fbiterm comes with its own fallback font
+
+	return 0
+}
+
+dialog_out=`mktemp -qt 'firstboot-XXXXXX'`
+d(){
+	retval=
+	while true
+	do
+		retval=0
+		dialog --backtitle "$PRETTY_NAME" --output-fd 3 "$@" 3>"${dialog_out}" || retval=$?
+		case $retval in
+		  0)
+			# need || true as dialog doesn't write newlines
+			read result < $dialog_out || true
+			return 0
+			;;
+		  1)
+			echo "$(xargs -a "$dialog_out")" >/var/log/jeos
+			dialog --yesno $"Do you really want to quit?" 0 0 && exit 1
+			continue
+			;;
+		  255)
+			# xargs to remove whitespaces
+			echo "$(xargs -a "$dialog_out")" >/var/log/jeos
+			result_error="$(xargs -a "$dialog_out")"
+			if [ -z "$result_error" ]; then
+				dialog --yesno $"Do you really want to quit?" 0 0 && exit 1
+				continue
+			fi
+			logger -p err -t jeos-firstboot "$result_error"
+			dialog --msgbox $"Exiting due to error, please check the system log" 0 0
+			exit 2
+			;;
+		esac
+	done
+}
+
+warn(){
+	d --title $"Warning" --msgbox "$1" 6 40
+}
+
+# localectl --no-pager list-keymaps does not list aliases (symlinks), but those are used
+# by YaST/langset.sh, so we need to show them.
+findkeymaps()
+{
+        list=()
+        local line
+        while read line; do
+                list+=("${line%.map.gz}" '')
+        done < <(find /usr/share/kbd/keymaps -name '*.map.gz' -printf "%f\n" | sort -u)
+        [ -n "$list" ]
+}
+
+findlocales()
+{
+        list=()
+        local l locale
+        # List only locales which are both in live-langset-data and glibc-locale(-base)
+        for l in /usr/share/langset/*; do
+                locale="${l#/usr/share/langset/}"
+                [ -d "/usr/lib/locale/${locale}.utf8" ] || continue
+                list+=("${locale}" '')
+        done
+        [ -n "$list" ]
+}
+
+get_current_locale()
+{
+	JEOS_LOCALE=`awk -F= '$1 == "LANG" { print $2; exit }' /etc/locale.conf`
```

Comment:
This should include a fallback to `en_US` and just echo the result instead of setting a variable.